### ISSUE TITLE
Which variable to use for Confusion Matrix

### ISSUE THREAD

**Issue by nlrahimi** [wuyuebupt/LargeScaleIncrementalLearning#4]
Hi,

Thank you for sharing the code in community.

I used this code on another dataset (11 classes only) and get high accuracy results, but when I generated a confusion matrix in resnet.py code the accuracy plummets drastically and all samples are pushed to one class. To make the confusion matrix, I added "probability" to predictions dictionary (line 522 in resnet)
`predictions = {
          'classes': tf.argmax(pred_all_classes, axis=1), \n
          'classes_before': tf.argmax(pred_all_classes_before, axis=1), \n
          **'probabilities': tf.nn.softmax(pred_all_classes, name='softmax_tensor'),**
      }`

I am not sure if "pred_all_classes" is the final test result because of the low accuracy it generates. But could not find which tensor the code is using that generates high accuracy predictions either.
I would appreciate if you can guide me in this and let me know which variable to use to generate the confusion matrix

Bests,
Nilla

**Comment by wuyuebupt** [comment]
Hi, thanks for interested in our work. 

I believe that you are using the correct one. 

For accuracy, the default metric is top-5 accuracy, e.g. https://github.com/wuyuebupt/LargeScaleIncrementalLearning/blob/master/resnet.py#L533. I am not sure if this is the reason that you get high accuracy. As you only have 11 classes, the top-5 could be high. 

Anyway, I would suggest that you check the accuracy again and see if the probability results match the auto-evaluated ones. They should be consisitent.

**Comment by nlrahimi** [comment]
That's right the  top 5 classes have high accuracy. though for the rest of classes it will not go any less than %80.
Yes the probability results matched the pred_all_class matrix, but its not a match for the code's auto-evaluated probabilities that results to high accuracy. That is why I thought maybe "pred_all_classes" is not final test results after the bias correction. 
But anyways I will not include the confusion matrix, and will close this issue. 
Thank you for the reply

Keep up the great work,
Bests.
